### ISSUE TITLE
24 -> 25 migration is broken on mainnet archival data

### ISSUE THREAD

**Issue by bowenwang1996** [near/nearcore#4786]
To reproduce, simply restart an archival node on mainnet with latest master commit.
```
thread 'main' panicked at 'source slice length (0) does not match destination slice length (62)', /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/core/src/slice/mod.rs:3014:13
```

**Comment by bowenwang1996** [comment]
This blocks the 1.21.0 release

**Comment by janewang** [comment]
Does this imply the release on Monday is delayed?

**Comment by bowenwang1996** [comment]
> Does this imply the release on Monday is delayed?

Yes

**Comment by frol** [comment]
@bowenwang1996 we run Indexer for Explorer based on 1.21.0-rc.1 release for mainnet over archival data and it migrated the database just fine today :thinking: 

I believe the problem is somewhere after 1.21.0-rc.1 release and master.

**Comment by bowenwang1996** [comment]
@frol I see. Thanks for point this out!

**Comment by janewang** [comment]
Thank you @frol!

@bowenwang1996 If the error is one off, we are good with 1.21.0 release as planned on Monday?

**Comment by bowenwang1996** [comment]
> @bowenwang1996 If the error is one off, we are good with 1.21.0 release as planned on Monday?

Yeah we should still be able to proceed with the new release

**Comment by nearmax** [comment]
@janewang , @bowenwang1996 as we discussed, we should have an empty release in this case to maintain the cadence.

**Comment by ailisp** [comment]
This kind of error doesn't happen at the time I'm testing the last fix. Means either a commit between 0130ae8a1de8484730fda169a678d190d605b7ae and master breaks something or mainnet archival during this month has accumulated some other kind of invalid data. Let me investigate

**Comment by ailisp** [comment]
So far I found it fails at a quite unfortunate place:
```
        if Vec::<ExecutionOutcomeWithIdAndProof>::try_from_slice(&value).is_ok() {
```

In the migration. This line, borsh should returns an error, and we'll enter one of the error handling branch below. but it panicked (!).

**Comment by ailisp** [comment]
I figured out, it is due to https://github.com/near/nearcore/pull/4438 merged after. #4438 cause this panic when deserialization fail. I'm testing a fix, and will post a PR shortly

**Comment by ailisp** [comment]
Fixed in #4790

**Comment by bowenwang1996** [comment]
> @janewang , @bowenwang1996 as we discussed, we should have an empty release in this case to maintain the release cadence. Also, this is something that was requested by @ilblackdragon .

Both frol and I tested the release. It turned out this issue did not affected the release. See https://github.com/near/nearcore/issues/4786#issuecomment-912805450
